scripts/add_nutriscore_to_scanbot_csv.pl

```diff
@@ -76,21 +105,39 @@
 		$found_unique_scans = $unique_scans;
 		
 		my $product_ref = retrieve_product($code);
-		if ((defined $product_ref) and (defined $product_ref->{nutriscore_grade})
-		and ($product_ref->{nutriscore_grade} =~ /^[a-e]$/)) {
-			$nutriscore_scans = $scans;
-			$nutriscore_unique_scans = $unique_scans;
-			$nutriscore_status = 1;
-			$total{nutriscore_products}++;
+		if (defined $product_ref) {
+
+			if ((defined $product_ref->{nutriscore_grade})
+				and ($product_ref->{nutriscore_grade} =~ /^[a-e]$/)) {
+				$nutriscore_scans = $scans;
+				$nutriscore_unique_scans = $unique_scans;
+				$nutriscore_status = 1;
+				$total{nutriscore_products}++;
+			}
+			if ((defined $product_ref->{ecoscore_grade})
+                                and ($product_ref->{ecoscore_grade} =~ /^[a-e]$/)) {
+                                $ecoscore_scans = $scans;
+                                $ecoscore_unique_scans = $unique_scans;
+                                $ecoscore_status = 1;
+                                $total{ecoscore_products}++;
+                        }
```

Comment:
style: your indentation is a bit strange (not consistent) :-)

Reply:
that's because I made the changes in prod with vi ;)

fixed.